frontend/src/components/dashboard/contactsHomePage/ContactHomePage.js

```diff
@@ -0,0 +1,146 @@
+import React, { Component } from "react";
+import { bindActionCreators, compose } from "redux";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+import queryString from "query-string";
+import { openModal, closeModal } from "@/actions/modalActions";
+import CustomPropTypes from "@/customPropTypes";
+import { fetchParties } from "@/actionCreators/partiesActionCreator";
+import { AuthorizationGuard } from "@/HOC/AuthorizationGuard";
+import * as Permission from "@/constants/permissions";
+import { getParties, getPartyIds, getPartyPageData } from "@/selectors/partiesSelectors";
+import ContactList from "@/components/dashboard/contactsHomePage/ContactList";
+import ResponsivePagination from "@/components/common/ResponsivePagination";
+import Loading from "@/components/common/Loading";
+import * as Strings from "@/constants/strings";
+import * as router from "@/constants/routes";
+
+/**
+ * @class ContactHomePage is the main landing page of the application, currently contains a List and Map View, ability to create a new mine, and search for a mine by name or lat/long.
+ *
+ */
+
+const propTypes = {
+  fetchParties: PropTypes.func.isRequired,
+  openModal: PropTypes.func.isRequired,
+  closeModal: PropTypes.func.isRequired,
+  parties: PropTypes.arrayOf(CustomPropTypes.party).isRequired,
+  partyIds: PropTypes.arrayOf(PropTypes.string).isRequired,
+  pageData: PropTypes.objectOf(CustomPropTypes.partyPageData).isRequired,
+};
+
+export class ContactHomePage extends Component {
+  state = {
+    isContactList: false,
+    params: {
+      page: Strings.DEFAULT_PAGE,
+      per_page: Strings.DEFAULT_PER_PAGE,
+    },
+  };
+
+  componentDidMount() {
+    const params = this.props.location.search;
+    if (params) {
+      this.renderDataFromURL(params);
+    } else {
+      this.props.history.push(
+        router.CONTACT_HOME_PAGE.dynamicRoute({
+          page: Strings.DEFAULT_PAGE,
+          per_page: Strings.DEFAULT_PER_PAGE,
+        })
+      );
+    }
+    this.props.fetchParties(params).then(() => {
+      this.setState({ isContactList: true });
+    });
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const locationChanged = nextProps.location !== this.props.location;
+    if (locationChanged) {
+      const params = nextProps.location.search;
+      this.renderDataFromURL(params);
+    }
+  }
+
+  componentWillUnmount() {
+    this.setState({ params: {} });
+  }
+
+  renderDataFromURL = (params) => {
+    const { page, per_page } = queryString.parse(params);
```

Comment:
It seems to me like `params` should be the object, not the query string. That way we can pass around the params object in the code, and then stringify it with `queryString.stringify` in the route function (as mentioned in another comment)